port/common/omrmemtag.c

```diff
@@ -285,18 +289,26 @@ omrmem_reallocate_memory(struct OMRPortLibrary *portLibrary, void *memoryPointer
 		omrmem_free_memory(portLibrary, memoryPointer);
 	} else {
 		memoryPointer = unwrapBlockAndCheckTags(portLibrary, memoryPointer);
+		savedMemTag = *((J9MemTag*)memoryPointer);
 		if (NULL == callSite) {
 			/* Inherit the callsite from the original allocation */
-			callSite = ((J9MemTag *) memoryPointer)->callSite;
+			callSite = savedMemTag.callSite;
```

Comment:
Restored to the original state.
```suggestion
			callSite = ((J9MemTag *) memoryPointer)->callSite;
```